controllers/spiaccesstokenbinding_controller.go

```diff
@@ -308,19 +356,46 @@ func (r *SPIAccessTokenBindingReconciler) Reconcile(ctx context.Context, req ctr
 	binding.Status.OAuthUrl = token.Status.OAuthUrl
 	binding.Status.UploadUrl = token.Status.UploadUrl
 
-	existingSyncedSecretName := ""
+	secretName := binding.Status.SyncedObjectRef.Name
+	serviceAccountName := binding.Status.ServiceAccountName
+
 	if token.Status.Phase == api.SPIAccessTokenPhaseReady {
-		ref, err := r.syncSecret(ctx, sp, &binding, token)
+		// linking the service account with a secret is a 3-step process.
+		// First, an empty service account needs to be created.
+		// Second, a secret linking to the service account needs to be created.
+		// Third, the service account needs to be updated with the link to the secret.
+
+		// TODO this can leave the SA behind if we subsequently fail to update the status of the binding
+		serviceAccount, err := r.getOrCreateServiceAccount(ctx, &binding)
+		if err != nil {
+			lg.Error(err, "unable to sync the service account")
+			return ctrl.Result{}, fmt.Errorf("failed to sync the service account: %w", err)
+		}
+		if serviceAccount != nil {
+			binding.Status.ServiceAccountName = serviceAccount.Name
+			lg.Info("service account linked", "saName", serviceAccount.Name, "managed", binding.Spec.ServiceAccount.Managed)
+		}
+
+		// TODO this can leave the secret behind if we subsequently fail to update the status of the binding
```

Comment:
is issue created for this TODO?

Reply:
https://issues.redhat.com/browse/SVPI-210